src/appleseed/renderer/kernel/shading/shadingpoint.cpp

```diff
@@ -808,22 +811,36 @@ void ShadingPoint::compute_shading_basis() const
         tangent = m_assembly_instance_transform.vector_to_parent(tangent);
     }
 
-    // Construct an orthonormal basis.
-    const Vector3d& sn = get_original_shading_normal();
-    const Vector3d bitangent = cross(tangent, sn);
-    const double norm_bitangent = norm(bitangent);
-    if (norm_bitangent >= 1.0e-6)
+
+    if (m_primitive_type == PrimitiveCurve3)
```

Comment:
What about `PrimitiveCurve1`?

Reply:
I am still not sure what the exact purpose of linear curves are, but they would not be compatible with the hair shader so I do not calculate their shading basis differently.

Reply:
> I am still not sure what the exact purpose of linear curves are

That should be discussed indeed. You are the curves/hair expert so please tell us if you think support for linear curves is useless and should be removed :) Do other renderers have them?